src/Operation/FindAndModify.php

```diff
@@ -253,7 +253,7 @@ public function execute(Server $server)
 
         $result = current($cursor->toArray());
 
-        return $result->value ?? null;
+        return is_object($result) ? $result->value ?? null : null;
```

Comment:
I know PHP doesn't require it, but this helps readability without having to remember operator precedence rules.

```suggestion
        return is_object($result) ? ($result->value ?? null) : null;
```

Alternatively, since this is one of many cases where psalm presumably doesn't like us using `->` on potential non-objects within an `isset()` context, perhaps there's a config option that we can use to ignore this? I don't feel strongly about it, so up to you if you're comfortable with the extra `is_object` calls.

Reply:
Good point. We could technically ignore the error and add it to the baseline, deferring it to later when we handle type maps correctly. For now I applied your suggestion to make the code more readable 👍